src/test/starknet.t.sol

```diff
@@ -154,4 +161,48 @@ contract StarknetTests is DssSpellTestBase, ConfigStarknet {
 
         assertTrue(core.isNotFinalized());
     }
+
+    function _checkStarknetMessage(bytes32 _spell) internal {
+        StarknetCoreLike core = StarknetCoreLike(addr.addr("STARKNET_CORE"));
```

Comment:
Why not test the GovRelay itself? I mean instead of replicating `relay` method logic it can be called directly using forge prank mode.

Reply:
This test happens after the spell is cast in the setup, so attempting to re-relay here will result in a different nonce and hash.

Reply:
You are right. I misread your test, I was under impression that it tries to send l1Tol2 message. 
Still, I think that it test internal details of Starknet core contract which is not necessary. Contract of `sendMessageToL2` method is that if it does not revert, message is queued.